Main.py

```diff
@@ -142,7 +143,69 @@ def main(args, seed=None, baked_server_options: Optional[Dict[str, object]] = No
 
     AutoWorld.call_all(world, "generate_basic")
 
-    # temporary home for item links, should be moved out of Main
+    # temporary home for custom item pool and item links, should be moved out of Main
+    for player in world.player_ids:
+        if world.custom_item_pool[player]:
+            player_item_pool = {}
+            custom_item_pool = {}
+            for item in world.itempool:
```

Comment:
if this is being called before fill why are you looking at the entire itempool instead of just the current player's itempool?

Reply:
I'm not sure what exactly you're suggesting. Worlds add items for their games directly into world.itempool - I'm not sure how else I would be looking at players' items